src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,126 @@
+
+                } else if (message.startsWith("delete")) {
+                    taskList.deleteTask(taskList.taskToDelete(message));
+                    storage.updateFile(taskList.getTasks());
+
+                } else if (message.startsWith("done")) {
+                    taskList.markAsCheckedTask(taskList.taskToCheck(message));
+                    storage.updateFile(taskList.getTasks());
+
+                } else if (message.startsWith("find")) {
+                    String keyword = message.substring(5);
+                    taskList.findTask(keyword);
+
+                }else if (message.startsWith("todo")) {
+                    isValidEntry(message, "todo");
+                    String taskName = message.substring(message.indexOf(" "));
+                    taskList.addTask(new Todo(taskName, false));
+                    storage.updateFile(taskList.getTasks());
+
+                }else if (message.startsWith("deadline")) {
+                    isValidEntry(message, "deadline");
+                    isFormatCorrect(message, "deadline");
+                    String taskName = message.substring(message.indexOf(" "), message.indexOf("/"));
+                    String temp = message.substring(message.indexOf("/") + 1);
+                    String due = temp.substring(temp.indexOf(" ") + 1);
+                    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+                    LocalDateTime parsedDate = LocalDateTime.parse(due, formatter);
+                    taskList.addTask(new Deadline(taskName, parsedDate, false));
+                    storage.updateFile(taskList.getTasks());
+
+                }else if (message.startsWith("event")) {
+                    isValidEntry(message, "event");
+                    isFormatCorrect(message, "event");
+                    String taskName = message.substring(message.indexOf(" "), message.indexOf("/"));
+                    String temp = message.substring(message.indexOf("/") + 1);
+                    String due = temp.substring(temp.indexOf(" ") + 1);
+                    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+                    LocalDateTime parsedDate = LocalDateTime.parse(due, formatter);
+                    taskList.addTask(new Event(taskName, parsedDate, false));
+                    storage.updateFile(taskList.getTasks());
+
+                }else {
+                    notValid();
+                }
+            } catch (DukeException | IOException e) {
+                System.out.println(e.getMessage());
+
+            } catch (DateTimeParseException e) {
+                String errorMessage = "Format of date and time should be \n yyyy-MM-dd HH:mm";
+                System.out.println(errorMessage + "\n Please try again.");
+            }
+        }
+        sc.close();
+    }
+
+    public void isValidEntry(String message, String type) throws DescriptionEmptyException {
```

Comment:
Maybe a bit nitpicky, but perhaps the method names could be checkValidEntry, checkCorrectFormat, throwNotValid or something along those lines since methods should be verbs.